### PR TITLE
JAX: Sync with master branch & resolve conflicts

### PR DESCRIPTION
**Pull request by AnirudhDagar**
@astonzhang I've fixed the merge conflicts and also added JAX code for the changes that Alex made recently in #2371
You can focus only on the [merge commit](https://github.com/d2l-ai/d2l-en/pull/2379/commits/2b11e2855edacc7ae98636be38e38bac415de4cf) to see the changes I've made. Once we merge this to JAX branch, we can then directly merge JAX branch back to master without worrying about any conflicts.

I request you to hold on to any changes to master branch now. Thanks! 

Files that needed conflict resolution:
```bash
	deleted by them: chapter_attention-mechanisms-and-transformers/attention-cues.md
	both modified:   chapter_attention-mechanisms-and-transformers/attention-pooling.md
	both modified:   chapter_convolutional-modern/cnn-design.md
	both modified:   chapter_multilayer-perceptrons/mlp-implementation.md
	both modified:   chapter_recurrent-modern/bi-rnn.md
	both modified:   chapter_recurrent-modern/gru.md
	both modified:   chapter_recurrent-modern/seq2seq.md
	both modified:   static/build.yml
```

New files added
```bash
new file:   chapter_attention-mechanisms-and-transformers/queries-keys-values.md
```

This PR also adds JAX code for the new file `queries-keys-values.md` and JAX code for rewritten `attention-pooling` section.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_attention-mechanisms-and-transformers/attention-pooling.md [hunk 1]
```diff
@@ -1,21 +1,22 @@
-# Attention Pooling
+# Attention Pooling by Similarity
```
Comment (AnirudhDagar) [inline chapter_attention-mechanisms-and-transformers/attention-pooling.md:1]: @astonzhang changes made to this file.

File: chapter_attention-mechanisms-and-transformers/attention-cues.md [hunk 2]
```diff
@@ -1,255 +0,0 @@
-```{.python .input}
```
Comment (AnirudhDagar) [inline chapter_attention-mechanisms-and-transformers/attention-cues.md:1]: @astonzhang changes made to this file.

File: chapter_attention-mechanisms-and-transformers/queries-keys-values.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
+```{.python .input  n=1}
```
Comment (AnirudhDagar) [inline chapter_attention-mechanisms-and-transformers/queries-keys-values.md:1]: @astonzhang changes made to this file.

File: chapter_convolutional-modern/cnn-design.md [hunk 4]
```diff
@@ -6,140 +6,52 @@ tab.interact_select(['mxnet', 'pytorch', 'tensorflow', 'jax'])
 # Designing Convolution Network Architectures
```
Comment (AnirudhDagar) [inline chapter_convolutional-modern/cnn-design.md:6]: @astonzhang changes made to this file.

File: chapter_recurrent-modern/lstm.md [hunk 5]
```diff
@@ -330,13 +330,27 @@ returns the final state and the stacked outputs as expected.
 %%tab pytorch, mxnet, tensorflow
 @d2l.add_to_class(LSTMScratch)
 def forward(self, inputs, H_C=None):
-    H, C = None, None if H_C is None else H_C
+    if H_C is None:
```
Comment (AnirudhDagar) [inline chapter_recurrent-modern/lstm.md:333]: @astonzhang changes made to this file.

**Review by AnirudhDagar — COMMENTED** [review]
I've commented out the files that were conflicting earlier and needed change for help in review.

File: static/build.yml [hunk 6]
```diff
@@ -8,8 +8,8 @@ dependencies:
     - -f https://download.pytorch.org/whl/torch_stable.html
     - torchvision==0.13.0+cu102
     - -f https://download.pytorch.org/whl/torch_stable.html
-    - tensorflow==2.9.1
-    - tensorflow-probability==0.17.0
+    - tensorflow==2.11.0
```
Comment (AnirudhDagar) [inline static/build.yml:11]: @astonzhang changes made to this file.

File: chapter_recurrent-modern/bi-rnn.md [hunk 7]
```diff
@@ -118,7 +118,7 @@ from d2l import jax as d2l
 from jax import numpy as jnp
 ```
 
-### Implementation from Scratch
+## Implementation from Scratch
```
Comment (AnirudhDagar) [inline chapter_recurrent-modern/bi-rnn.md:121]: @astonzhang changes made to this file.

**Comment by astonzhang** [comment]
Thanks. I've reviewed all the individual files that you commented. Note that we'll need JAX implementation for attention-scoring-functions, bahdanau-attention, etc later.

**Pull request merged**